### PR TITLE
Automate first four affiliated cert tests

### PR DESCRIPTION
**Pull request by theemj**

### CODE REVIEW

File: tests/affiliatedcertification/affiliatedcertparameters/paramaters.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package affiliatedcertparameters
+
+var (
+	AffiliatedCertificationTestSuiteName = "affiliated-certification"
+	TestCaseContainerSkipRegEx           = "operator-is-certified"
+	TestCaseOperatorSkipRegEx            = "container-is-certified"
+	TestCaseContainerAffiliatedCertName  = "affiliated-certification affiliated-certification-container-is-certified"
+	CertifiedContainer1                  = "nodejs-12/ubi8"
```
Comment (kononovn) [inline tests/affiliatedcertification/affiliatedcertparameters/paramaters.go:8]: Please re-name this var from CertifiedContainer1 to CertifiedContainerNodeJsUbi

File: tests/affiliatedcertification/affiliatedcertparameters/paramaters.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+	CertifiedContainer2                  = "openjdk-11-rhel7/openjdk"
```
Comment (kononovn) [inline tests/affiliatedcertification/affiliatedcertparameters/paramaters.go:9]: Please rename this var

File: tests/affiliatedcertification/affiliatedcertparameters/paramaters.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+	UncertifiedContainer1                = "foo/bar"
```
Comment (kononovn) [inline tests/affiliatedcertification/affiliatedcertparameters/paramaters.go:10]: Please re-name this var

File: tests/affiliatedcertification/tests/affiliated_certification_container.go [hunk 4]
```diff
@@ -17,22 +23,94 @@ var _ = Describe("Affiliated-certification container certification,", func() {
 
 	// 46562
 	It("one container to test, container is certified", func() {
-		Skip("Under development")
+		By("Add container information to tnf_config.yml")
```
Comment (kononovn) [inline tests/affiliatedcertification/tests/affiliated_certification_container.go:26]: Please remove hardcoded tnf_config.yml and use one from globalparameters.go

File: tests/affiliatedcertification/tests/affiliated_certification_container.go [hunk 5]
```diff
@@ -17,22 +23,94 @@ var _ = Describe("Affiliated-certification container certification,", func() {
...
+		err := globalhelper.DefineTnfConfig(
+			[]string{netparameters.TestNetworkingNameSpace},
+			[]string{netparameters.TestPodLabel},
+			[]string{affiliatedcertparameters.CertifiedContainer1})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Start tests")
```
Comment (kononovn) [inline tests/affiliatedcertification/tests/affiliated_certification_container.go:33]: Should be "Start test"

File: tests/affiliatedcertification/tests/affiliated_certification_container.go [hunk 6]
```diff
@@ -17,22 +23,94 @@ var _ = Describe("Affiliated-certification container certification,", func() {
...
+		err = nethelper.ValidateIfReportsAreValid(
+			affiliatedcertparameters.TestCaseContainerAffiliatedCertName,
+			globalparameters.TestCasePassed)
+		Expect(err).ToNot(HaveOccurred())
 	})
 
 	// 46563
 	It("one container to test, container is not certified [negative]", func() {
-		Skip("Under development")
+		By("Add container information to tnf_config.yml")
```
Comment (kononovn) [inline tests/affiliatedcertification/tests/affiliated_certification_container.go:49]: Please remove hardcoded tnf_config.yml and use one from globalparameters.go

File: tests/affiliatedcertification/tests/affiliated_certification_container.go [hunk 7]
```diff
@@ -17,22 +23,94 @@ var _ = Describe("Affiliated-certification container certification,", func() {
...
+		err := globalhelper.DefineTnfConfig(
+			[]string{netparameters.TestNetworkingNameSpace},
+			[]string{netparameters.TestPodLabel},
+			[]string{affiliatedcertparameters.UncertifiedContainer1})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Start tests")
```
Comment (kononovn) [inline tests/affiliatedcertification/tests/affiliated_certification_container.go:56]: test

File: tests/affiliatedcertification/tests/affiliated_certification_container.go [hunk 8]
```diff
@@ -17,22 +23,94 @@ var _ = Describe("Affiliated-certification container certification,", func() {
 
 	// 46562
 	It("one container to test, container is certified", func() {
-		Skip("Under development")
+		By("Add container information to tnf_config.yml")
+		err := globalhelper.DefineTnfConfig(
+			[]string{netparameters.TestNetworkingNameSpace},
+			[]string{netparameters.TestPodLabel},
+			[]string{affiliatedcertparameters.CertifiedContainer1})
+		Expect(err).ToNot(HaveOccurred())
```
Comment (kononovn) [inline tests/affiliatedcertification/tests/affiliated_certification_container.go:31]: Let's add more logs. Please add optionalDescritpion for every expect call.
For example: Expect(err).ToNot(HaveOccurred(), "Error to define tnf config file")

File: tests/affiliatedcertification/tests/affiliated_certification_container.go [hunk 9]
```diff
@@ -17,22 +23,94 @@ var _ = Describe("Affiliated-certification container certification,", func() {
...
 	It("two containers to test, one is certified, one is not [negative]", func() {
-		Skip("Under development")
+		By("Add container information to tnf_config.yml")
+		err := globalhelper.DefineTnfConfig(
+			[]string{netparameters.TestNetworkingNameSpace},
+			[]string{netparameters.TestPodLabel},
+			[]string{affiliatedcertparameters.UncertifiedContainer1, affiliatedcertparameters.CertifiedContainer1})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Start tests")
```
Comment (kononovn) [inline tests/affiliatedcertification/tests/affiliated_certification_container.go:102]: test

File: tests/globalhelper/helper.go [hunk 10]
```diff
@@ -61,8 +61,32 @@ func defineTargetPodLabels(config *globalparameters.TnfConfig, targetPodLabels [
 	return nil
 }
 
+func defineCertifiedContainerInfo(config *globalparameters.TnfConfig, certifiedContainerInfo []string) error {
```
Comment (kononovn) [inline tests/globalhelper/helper.go:64]: Please re-name to defineCertifiedContainersInfo

**Review by kononovn — CHANGES_REQUESTED** [review]

**Comment by kononovn** [comment]
lgtm

**Review by kononovn — APPROVED** [review]
Looks good to me, well done!

File: tests/globalhelper/helper.go [hunk 11]
```diff
@@ -61,8 +61,32 @@ func defineTargetPodLabels(config *globalparameters.TnfConfig, targetPodLabels [
...
+
+	for _, certifiedContainerFields := range certifiedContainerInfo {
+		nameRepository := strings.Split(certifiedContainerFields, "/")
+		if len(nameRepository) != 2 {
+			return fmt.Errorf(fmt.Sprintf("certified container info %s is invalid", certifiedContainerFields))
+		}
+
+		name := strings.TrimSpace(nameRepository[0])
+		repo := strings.TrimSpace(nameRepository[1])
+
```
Comment (sebrandon1) [inline tests/globalhelper/helper.go:78]: Might I suggest adding a logging line here with the `name` and `repo`.

**Review by sebrandon1 — APPROVED** [review]
One nit about a logging line but otherwise looks good!

**Comment by sebrandon1** [comment]
Thanks!

**Pull request merged**